File: projects/sse-webflux-servlet-projects/retry-server/src/main/java/com/example/retryserver/dto/OrderResponse.java
package com.example.retryserver.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponse {

	@JsonProperty("order_id")
	private Long orderId;

	@JsonProperty("user_id")
	private Long userId;

	@JsonProperty("total_amount")
	private BigDecimal totalAmount;

	@JsonProperty("order_status")
	private String orderStatus;

	@JsonProperty("item_count")
	private Integer itemCount;

	@JsonProperty("shipping_address")
	private String shippingAddress;

	@JsonProperty("order_date")
	private LocalDateTime orderDate;

	@JsonProperty("estimated_delivery")
	private LocalDateTime estimatedDelivery;

	@JsonProperty("payment_method")
	private String paymentMethod;

	public static OrderResponse of(Long id) {
		return new OrderResponse(
			id,
			100L + id,
			new BigDecimal("99." + String.format("%02d", (id % 100))),
			"PROCESSING",
			(int) ((id % 10) + 1),
			id + " Main St, City, State " + String.format("%05d", id),
			LocalDateTime.now().minusDays(id),
			LocalDateTime.now().plusDays(5 + (id % 5)),
			"CREDIT_CARD"
		);
	}
}

File: projects/spring-statemachine-poc/src/main/kotlin/com/example/statemachine/domain/OrderResponse.kt
package com.example.statemachine.domain

import java.math.BigDecimal
import java.time.LocalDateTime

/**
 * 주문 응답 DTO
 *
 * Controller에서 클라이언트에게 반환할 때 사용.
 * Entity를 직접 노출하지 않고 DTO로 변환하여 API 계약을 분리.
 */
data class OrderResponse(
    val id: Long,
    val state: OrderState,
    val customerName: String,
    val productName: String,
    val amount: BigDecimal,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)

/**
 * Order Entity → OrderResponse DTO 변환 확장 함수
 */
fun Order.toResponse() = OrderResponse(
    id = this.id ?: throw IllegalStateException("저장되지 않은 주문입니다"),
    state = this.state,
    customerName = this.customerName,
    productName = this.productName,
    amount = this.amount,
    createdAt = this.createdAt,
    updatedAt = this.updatedAt
)
